cluster-autoscaler/core/scaleup/orchestrator/async_initializer_test.go

```diff
@@ -106,6 +107,19 @@ func TestNodePoolAsyncInitialization(t *testing.T) {
 					FailedResizeNodeGroups: []cloudprovider.NodeGroup{createdNodeGroup},
 					PodsTriggeredScaleUp:   option.Pods,
 				})
+			} else {
```

Comment:
nit: instead of `if .. else` block here I'd extend the test case struct with a `wantStatus` field and just `assert.Equal(t, scaleUpStatusProcessor.lastStatus, tc.wantStatus)`. Making test cases explicit like this makes reading them easier.

Reply:
Right, it looks better now, but still a bit obfuscated by all the dependencies.